tests/phpunit/tests/html-api/wpHtmlTagProcessor-internals.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Tag_Processor internal helpers.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ */
+
+/**
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Tag_Processor
+ */
+class Tests_HtmlApi_WpHtmlTagProcessor_Internals extends WP_UnitTestCase {
+	/**
+	 * @ticket {TICKET NUMBER}
+	 *
+	 * @dataProvider data_html_with_entire_tag_raw_markup
+	 *
+	 * @param string $html_with_target_element HTML with a tag containing "target" attribute.
+	 * @param string $expected_raw_markup      Expect full raw markup for targeted tag.
+	 */
+	public function test_returns_raw_html_markup_for_entire_tag( $html_with_target_element, $expected_raw_markup ) {
+		$processor = new WP_HTML_Tag_Processor( $html_with_target_element );
+		while ( $processor->next_tag() && null == $processor->get_attribute( 'target' ) ) {
```

Comment:
Double `==` (flagged by linter). I guess we need plain ol'

```suggestion
		while ( $processor->next_tag() && null === $processor->get_attribute( 'target' ) ) {
```

?

Reply:
Thanks for noticing. Equalized in 7151083918